returns/contrib/pytest/plugin.py

```diff
@@ -13,6 +17,17 @@
     'alt',
 )
 
+_FunctionType = TypeVar('_FunctionType', bound=Callable)
+# _ResultBasedType represents `Callable[[Any], ResultBasedN[Any, Any, Any]]`
+# but we cannot import `ResultBasedN` here because we're omitting this file from
+# the coverage report and `coverage.py` will exclude the class' coverage
+# information if we import it here.
+_ResultBasedType = TypeVar('_ResultBasedType', bound=Callable)
```

Comment:
Can we use 

```python
if TYPE_CHECKING:
   _ResultCallableType = TypeVar('_ResultBasedType', bound=Callable[..., ResultBasedN])
```

?

Reply:
And then use it like `'_ResultCallableType'`

Reply:
The problem is the import statment, but following your idea we can do this:
```python
if TYPE_CHECKING:
    from returns.interfaces.specific.result import ResultBasedN

_ResultCallableType = TypeVar(
    '_ResultCallableType', bound=Callable[..., 'ResultBasedN'],
)
```

Reply:
Yes, that's exactly what I've meant 🙂